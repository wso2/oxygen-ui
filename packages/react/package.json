{
  "name": "@oxygen-ui/react",
  "version": "2.3.4",
  "description": "The React implementation of the Oxygen Design System",
  "author": "WSO2",
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/index.d.ts",
  "homepage": "https://github.com/wso2/oxygen-ui/tree/main/packages/react#readme",
  "bugs": {
    "url": "https://github.com/wso2/oxygen-ui/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/wso2/oxygen-ui",
    "directory": "packages/react"
  },
  "keywords": [
    "react",
    "react-library",
    "design-system",
    "oxygen-ui",
    "oxygen-ui-react"
  ],
  "scripts": {
    "build": "rollup -c",
    "build:watch": "nodemon --watch src --ext js,ts,jsx,tsx,mjs,json --exec \"rollup -c && pnpm run postbuild\"",
    "build:storybook": "build-storybook --no-manager-cache",
    "prebuild": "rimraf dist",
    "postbuild": "node scripts/build-components.js",
    "lint": "pnpm lint:es && pnpm lint:styles",
    "lint:es": "eslint . --ext .js,.jsx,.ts,.tsx",
    "lint:styles": "stylelint 'src/**/*.{css,scss,sass}' --config stylelint.config.cjs --allow-empty-input",
    "start": "pnpm storybook",
    "storybook": "start-storybook -p 6006",
    "test": "NODE_OPTIONS=--experimental-vm-modules pnpm jest --passWithNoTests",
    "typecheck": "tsc --noemit -p tsconfig.json"
  },
  "dependencies": {
    "@mui/x-data-grid": "^6.9.2",
    "@oxygen-ui/primitives": "2.3.4",
    "@oxygen-ui/react-icons": "2.3.4",
    "clsx": "^1.2.1",
    "react-world-flags": "^1.5.1"
  },
  "devDependencies": {
    "@babel/core": "^7.20.2",
    "@emotion/react": "^11.10.5",
    "@emotion/styled": "^11.10.5",
    "@mui/icons-material": "^5.10.16",
    "@mui/lab": "5.0.0-alpha.110",
    "@mui/material": "^5.10.16",
    "@mui/system": "^5.10.16",
    "@mui/utils": "^5.10.16",
    "@oxygen-ui/logger": "2.3.4",
    "@rollup/plugin-commonjs": "^23.0.3",
    "@rollup/plugin-image": "^3.0.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "@rollup/plugin-typescript": "^10.0.1",
    "@storybook/addon-a11y": "^6.5.13",
    "@storybook/addon-actions": "^6.5.13",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-interactions": "^6.5.13",
    "@storybook/addon-links": "^6.5.13",
    "@storybook/builder-webpack4": "^6.5.13",
    "@storybook/builder-webpack5": "^6.5.13",
    "@storybook/client-api": "^6.5.13",
    "@storybook/manager-webpack4": "^6.5.13",
    "@storybook/manager-webpack5": "^6.5.13",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/react": "^6.5.13",
    "@storybook/testing-library": "^0.0.13",
    "@storybook/types": "7.0.0-alpha.44",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@types/jest": "^29.2.3",
    "@types/node": "^18.11.18",
    "@types/react": "^18.0.25",
    "@types/react-dom": "^18.0.9",
    "@types/testing-library__jest-dom": "^5.14.5",
    "@wso2/eslint-plugin": "https://gitpkg.now.sh/brionmario/wso2-ui-configs/packages/eslint-plugin?b21bbbd7379ab3b7315d7d9478dbd88c0bf11241",
    "@wso2/prettier-config": "https://gitpkg.now.sh/brionmario/wso2-ui-configs/packages/prettier-config?b21bbbd7379ab3b7315d7d9478dbd88c0bf11241",
    "@wso2/stylelint-config": "https://gitpkg.now.sh/brionmario/wso2-ui-configs/packages/stylelint-config?b21bbbd7379ab3b7315d7d9478dbd88c0bf11241",
    "babel-jest": "^29.3.1",
    "babel-loader": "^8.3.0",
    "eslint": "8.25.0",
    "eslint-plugin-mdx": "^2.0.5",
    "fs-extra": "^11.1.0",
    "jest": "29.0.3",
    "jest-environment-jsdom": "^29.4.1",
    "node-sass": "^8.0.0",
    "nodemon": "^2.0.22",
    "postcss": "8.4.16",
    "prettier": "^2.8.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^3.5.0",
    "rollup-plugin-dts": "^5.0.0",
    "rollup-plugin-postcss": "^4.0.2",
    "rollup-plugin-terser": "^7.0.2",
    "sass": "^1.56.1",
    "sass-loader": "^13.2.0",
    "storybook-addon-designs": "^6.3.1",
    "storybook-addon-themes": "^6.1.0",
    "storybook-dark-mode": "^1.1.2",
    "stylelint": "^15.1.0",
    "ts-dedent": "^2.2.0",
    "ts-jest": "^29.0.3",
    "tsconfig-paths-webpack-plugin": "^4.0.0",
    "tslib": "^2.4.1",
    "typescript": "^4.9.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.10.5",
    "@emotion/styled": "^11.10.5",
    "@mui/icons-material": "^5.10.16",
    "@mui/lab": "5.0.0-alpha.110",
    "@mui/material": "^5.10.16",
    "@mui/system": "^5.10.16",
    "@mui/utils": "^5.10.16",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0",
    "typescript": ">=4.0.0"
  },
  "peerDependenciesMeta": {
    "typescript": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "publishPath": "dist"
}
