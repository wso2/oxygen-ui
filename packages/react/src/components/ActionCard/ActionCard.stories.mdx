import {ArgsTable, Source, Story, Canvas, Meta} from '@storybook/addon-docs';
import ActionCard from './ActionCard.tsx';
import dedent from 'ts-dedent';
import StoryConfig from '../../../.storybook/story-config.ts';

export const meta = {
  component: ActionCard,
  title: StoryConfig.ActionCard.hierarchy
};

<Meta title={meta.title} component={meta.component} />

export const Template = args => <ActionCard {...args} />;

# ActionCard

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)

## Overview

Action cards can be used in overview pages or dashboards.

<Canvas>
  <Story name="Overview" args={{
    title: "Secure your account by adding an extra layer of security",
    description: "Configure additional authentications to sign in easily or to add an extra layer of security to your account.",
    actionText: "Setup MFA",
    onActionClick:() => {alert("Action clicked")},
    image: <img src="assets/images/action-card-image.svg" alt="action card" />,
    variant: "outlined"
  }}>
    {Template.bind({})}
  </Story>
</Canvas>

## Props
<ArgsTable story="Overview" />

## Usage

Import and use the `ActionCard` component in your components as follows.

<Source
  language="jsx"
  dark
  format
  code={dedent`
import ActionCard from '@oxygen-ui/react/ActionCard';\n
function Demo() {
  return (
    <ActionCard
      title="Action Card Title"
      description="Action Card Description."
      actionText="Action Card Action Text"
      onActionClick={() => {alert("Action clicked")}}
      image={<img src="assets/images/action-card-image.svg" alt="action card" />}
    />
  );
}`}
/>
