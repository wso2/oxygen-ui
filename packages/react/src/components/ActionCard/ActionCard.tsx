/**
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import clsx from 'clsx';
import {FC, ReactElement, ReactNode} from 'react';
import {WithWrapperProps} from '../../models';
import {composeComponentDisplayName} from '../../utils';
import Button from '../Button';
import Card, {CardProps} from '../Card';
import CardActions from '../CardActions';
import CardContent from '../CardContent';
import Typography from '../Typography';
import './action-card.scss';

export interface ActionCardProps extends CardProps {
  /**
   * The text to be displayed in the action button.
   */
  actionText: string;
  /**
   * The description of the card.
   */
  description: string;
  /**
   * The image to be displayed in the card.
   */
  image: ReactNode;
  /**
   * Callback method to be called when the action button is clicked.
   */
  onActionClick: () => void;
  /**
   * The title of the card.
   */
  title: string;
}

const COMPONENT_NAME: string = 'ActionCard';

const ActionCard: FC<ActionCardProps> & WithWrapperProps = ({
  className,
  image,
  title,
  description,
  actionText,
  onActionClick,
  ...rest
}: ActionCardProps): ReactElement => {
  const classes: string = clsx('oxygen-action-card', className);

  return (
    <Card className={classes} {...rest}>
      <CardContent>
        {image}
        <Typography variant="subtitle1">{title}</Typography>
        <Typography variant="body2">{description}</Typography>
      </CardContent>
      <CardActions>
        <Button onClick={onActionClick} variant="contained">
          {actionText}
        </Button>
      </CardActions>
    </Card>
  );
};

ActionCard.displayName = composeComponentDisplayName(COMPONENT_NAME);
ActionCard.muiName = COMPONENT_NAME;

export default ActionCard;
