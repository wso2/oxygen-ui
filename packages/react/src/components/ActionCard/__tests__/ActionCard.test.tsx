/**
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {render} from '@unit-testing';
import {ReactElement} from 'react';
import ActionCard from '../ActionCard';

const onActionClick: jest.Mock<any, any> = jest.fn();

const ActionCardTestComponent: ReactElement = (
  <ActionCard
    title="Secure your account by adding an extra layer of security"
    description={
      'Configure additional authentications to sign in easily or to ' +
      'add an extra layer of security to your account.'
    }
    actionText="Setup MFA"
    onActionClick={onActionClick}
    image={<img src="assets/images/action-card-image.svg" alt="action card" />}
  />
);

describe('ActionCard', () => {
  it('should render successfully', () => {
    const {baseElement} = render(ActionCardTestComponent);
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const {baseElement} = render(ActionCardTestComponent);
    expect(baseElement).toMatchSnapshot();
  });
});
