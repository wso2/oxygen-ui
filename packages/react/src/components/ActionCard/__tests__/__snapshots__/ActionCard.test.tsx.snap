// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCard should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root oxygen-card oxygen-action-card css-1e3jyt5-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root oxygen-card-content css-46bh2p-MuiCardContent-root"
      >
        <img
          alt="action card"
          src="assets/images/action-card-image.svg"
        />
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 oxygen-typography css-n2qzog-MuiTypography-root"
        >
          Secure your account by adding an extra layer of security
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2 oxygen-typography css-xq9ne1-MuiTypography-root"
        >
          Configure additional authentications to sign in easily or to add an extra layer of security to your account.
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing oxygen-card-actions css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation oxygen-button css-1w4rwqf-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          id=":r1:"
          tabindex="0"
          type="button"
        >
          Setup MFA
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
